codec/src/main/java/io/netty/handler/codec/base64/Base64.java

```diff
@@ -362,43 +362,63 @@ public boolean process(byte value) throws Exception {
+                    // bits we care about.
+                    dest.setShort(destOffset, ((decodabet[src0] & 0x3f) << 2 | (b1 & 0xf0) >> 4) << 8 |
+                                              (b1 & 0xf) << 4 | (decodabet[src2] & 0xfc) >>> 2);
                 } else {
-                    dest.setShort(destOffset, (short) ((outBuff & 0xff0000) >>> 16 | (outBuff & 0xff00)));
+                    // This is just a simple byte swap of the operation above.
+                    dest.setShort(destOffset, (decodabet[src0] & 0x3f) << 2 | (b1 & 0xf0) >> 4 |
+                                              ((b1 & 0xf) << 4 | (decodabet[src2] & 0xfc) >>> 2) << 8);
                 }
                 return 2;
             }
 
             // Example: DkLE
-            final int outBuff;
             try {
-                outBuff = (decodabet[src[0]] & 0xff) << 18 |
-                          (decodabet[src[1]] & 0xff) << 12 |
-                          (decodabet[src[2]] & 0xff) <<  6 |
-                           decodabet[src[3]] & 0xff;
+                // Just directly set it as medium
+                if (dest.order() == ByteOrder.BIG_ENDIAN) {
+                    dest.setMedium(destOffset, (decodabet[src0] & 0x3f) << 18 |
+                                               (decodabet[src1] & 0xff) << 12 |
+                                               (decodabet[src2] & 0xff) << 6 |
+                                                decodabet[src3] & 0xff);
+                } else {
+                    final byte b1 = decodabet[src1];
+                    final byte b2 = decodabet[src2];
+                    // The goal is to byte swap the BIG_ENDIAN case above. There are 2 interesting things to consider:
+                    // 1. We are byte swapping a 3 byte data type. The top and the bottom byte switch, but the middle
+                    //    remains the same.
+                    // 2. The contents straddles byte boundaries. This means bytes will be pulled apart during the byte
+                    //    swapping process.
+                    dest.setMedium(destOffset, (decodabet[src0] & 0x3f) << 2 |
+                                            // The bottom half of b1 remains in the middle.
+                                               (b1 & 0xf) << 12 |
+                                            // The top half of b1 are the least significant bits after the swap.
+                                               (b1 & 0xf0) >>> 4 |
+                                            // The bottom 2 bits of b2 will be the most significant bits after the swap.
+                                               (b2 & 0x3) << 22 |
+                                            // The remaining 6 bits of b2 remain in the middle.
+                                               (b2 & 0xfc) << 6 |
+                                               (decodabet[src3] & 0xff) << 16);
+                }
             } catch (IndexOutOfBoundsException ignored) {
```

Comment:
good point ... it looks like we don't even enforce this uniformly in this method ... let me make it more consistent